volatility/framework/plugins/mac/bash.py

```diff
@@ -84,24 +83,23 @@ def _generator(self, tasks):
                 yield (0, (int(task.p_pid), task_name, hist.get_time_object(), hist.get_command()))
 
     def run(self):
-        filter_func = tasks.Tasks.create_pid_filter([self.config.get('pid', None)])
+        filter_func = pslist.PsList.create_pid_filter([self.config.get('pid', None)])
+        list_tasks = pslist.PsList.get_list_tasks(self.config.get('method', 'tasks'))
 
         return renderers.TreeGrid([("PID", int), ("Process", str), ("CommandTime", datetime.datetime),
                                    ("Command", str)],
                                   self._generator(
-                                      tasks.Tasks.list_tasks(self.context,
-                                                             self.config['primary'],
-                                                             self.config['darwin'],
-                                                             filter_func = filter_func)))
+                                      list_tasks(self.context,
+                                                 self.config['primary'],
+                                                 self.config['darwin'],
+                                                 filter_func = filter_func)))
 
     def generate_timeline(self):
-        filter_func = tasks.Tasks.create_pid_filter([self.config.get('pid', None)])
+        filter_func = pslist.PsList.create_pid_filter([self.config.get('pid', None)])
+        list_tasks = pslist.PsList.get_list_tasks(self.config.get('pslist_method', pslist.PsList.pslist_methods[0]))
```

Comment:
can pslist.PsList.pslist_methods[0] just be 'tasks' live above?

Also, why does this one do config.get('pslist_method', ...), but then the _generator does config.get('method',...) ?

Reply:
Well, making it the [0] element allows us to pick later by just having it in one place.  If we hard code it we'd have to change the order (the first should be the default) *and* we'd need to change the places it was hard coded.  Seems better to use the first element?

Not entirely sure about the pslist_method vs just method.  I think it might've been the first one, and I later decided it should say which method it's talking about.  Feel free to correct it, or leave this unresolved and I'll fix it when I next get a chance to code...  5:)